lib/cpl.rb

```diff
@@ -210,6 +215,10 @@ def self.all_base_commands
       ::Shell.abort("Unable to load command: #{e.message}")
     end
 
+    disable_required_check!(*@commands_with_required_options)
+    check_unknown_options!(except: @commands_with_extra_options)
+    stop_on_unknown_option!
```

Comment:
This ensures that an error is raised if unknown options are passed to any command, with the exception of commands that accept extra options and pass those forward.